cmd/broadcaster-cli/app/utxos/create/create.go

```diff
@@ -78,32 +77,42 @@ var Cmd = &cobra.Command{
 			return fmt.Errorf("failed to create client: %v", err)
 		}
 
+		names := helper.GetOrderedKeys(keySetsMap)
+
 		wocClient := woc_client.New(!isTestnet, woc_client.WithAuth(wocApiKey), woc_client.WithLogger(logger))
+		creators := make([]*broadcaster.UTXOCreator, 0, len(keySetsMap)) // Use the Creator interface for flexibility
+		for _, keyName := range names {
+			ks := keySetsMap[keyName]
+			creator, err := broadcaster.NewUTXOCreator(
+				logger.With(slog.String("address", ks.Address(!isTestnet)), slog.String("name", keyName)),
+				client, ks, wocClient, isTestnet, broadcaster.WithFees(miningFeeSat),
+			)
+			if err != nil {
+				return err
+			}
 
-		ks := make([]*keyset.KeySet, len(keySetsMap))
-		counter := 0
-		for _, keySet := range keySetsMap {
-			ks[counter] = keySet
-			counter++
-		}
-		rateBroadcaster, err := broadcaster.NewUTXOCreator(logger, client, ks, wocClient, isTestnet,
-			broadcaster.WithFees(miningFeeSat),
-		)
-		if err != nil {
-			return fmt.Errorf("failed to create broadcaster: %v", err)
+			creators = append(creators, creator)
 		}
 
-		err = rateBroadcaster.CreateUtxos(outputs, uint64(satoshisPerOutput))
-		if err != nil {
-			return fmt.Errorf("failed to create utxos: %v", err)
-		}
+		multiCreator := broadcaster.NewMultiKeyUTXOCreator(logger, creators)
```

Comment:
Where is the `Start()` function called?

Reply:
its here 
<img width="439" alt="Screenshot 2024-09-11 at 12 42 09" src="https://github.com/user-attachments/assets/c4477814-8551-4072-b136-30f232c20c81">

Reply:
acutally in this bit `multiCreator := broadcaster.NewMultiKeyUTXOCreator(logger, creators)

		signalChan := make(chan os.Signal, 1)
		signal.Notify(signalChan, os.Interrupt) // Listen for Ctrl+C

		// Graceful shutdown on interrupt signal
		go func() {
			<-signalChan
			multiCreator.Shutdown()
		}()

		logger.Info("Starting UTXO creation")
		multiCreator.Start(outputs, uint64(satoshisPerOutput))

		return nil`

Reply:
Ok, I see